transport/conn.go

```diff
@@ -360,6 +362,9 @@ func WrapConn(conn net.Conn) (*Conn, error) {
 		return c, err
 	}
 
+	c.w.info = c.String()
+	c.r.info = c.String()
+
```

Comment:
This can be moved to the declaration directly.

Reply:
That is under r: connReader{

Reply:
I'm afraid this can't be declared before running init() method which discovers shard. Also RemoteAddr() method in conn::String() causes nil pointer dereference when called in r: connReader{ or w: connWriter, since conn is not set.

Reply:
I was thinking about a function pointer not a string.